src/nupic/regions/PyRegion.py

```diff
@@ -177,6 +177,18 @@ def getOutputElementCount(self, name):
     name: the name of the output
     """
 
+  @not_implemented
+  def getAlgorithmInstance(self):
```

Comment:
I'm fine with this suggestion. The rest of the code samples will still work as-is.  If we do this we should add a doc string to the implementation for each individual Region.